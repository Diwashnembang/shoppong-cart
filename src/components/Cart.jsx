import React from 'react'
import CloseIcon from '@mui/icons-material/Close'

import './cart.css'

import Items from './Items'

const Cart = ({ items, setIsCartClicked }) => {
  /* sets is setIsCartClicked to false so that the app component doesnot render */
  /* cart  */
  const closeCart = () => {
    setIsCartClicked(false)
  }
  return (
    <div id='cart'>
      <div className='close-cart' onClick={closeCart} data-testid="close-cart">
        <CloseIcon fontSize='large' />
      </div>
      <div id='cart-title'>
        <h1>
          <span className='block'>Your</span>
          <span className='block'>Shopping</span>
          <span className='block'>Bag</span>
        </h1>
      </div>
      <div id='show-items' >
        {items.map((element, index) => (
          <div className='item' key={index} data-testid="show-items">
            <Items item={element} />
          </div>
        ))}
      </div>
    </div>
  )
}

export default Cart
