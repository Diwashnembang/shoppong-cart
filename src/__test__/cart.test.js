import {  render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import Cart from '../components/Cart'

 const mockItem = {
    id: "001",
    src: "src/assets/electronics/geforce-rtx-3080-ti.jpg",
    price: "400",
    name: "Graphic Card",
};

const mockSetIsCartClicked = jest.fn();

describe("Cart component",()=>{
  it("should render properly",()=>{
    const container=render(<Cart items={[mockItem]} setIsCartClicked={mockSetIsCartClicked}/>)
    expect(container).toMatchSnapshot();

  })

  it("should render 2 items",()=>{
   render(<Cart items={[mockItem,mockItem]} setIsCartClicked={mockSetIsCartClicked}/>);
    const items =screen.getAllByTestId('show-items');
    expect(items.length).toBe(2);
  })

  it("should called setIsCartClicked with false",()=>{
   render(<Cart items={[mockItem,mockItem]} setIsCartClicked={mockSetIsCartClicked}/>);
    const clickElement = screen.getByTestId('close-cart') ;
    userEvent.click(clickElement);
    expect(mockSetIsCartClicked).toBeCalledWith(false);

  })
})

