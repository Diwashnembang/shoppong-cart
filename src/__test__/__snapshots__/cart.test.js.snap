// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="cart"
      >
        <div
          class="close-cart"
          data-testid="close-cart"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          id="cart-title"
        >
          <h1>
            <span
              class="block"
            >
              Your
            </span>
            <span
              class="block"
            >
              Shopping
            </span>
            <span
              class="block"
            >
              Bag
            </span>
          </h1>
        </div>
        <div
          id="show-items"
        >
          <div
            class="item"
            data-testid="show-items"
          >
            <div
              class="selected-item"
            >
              <div
                class="selectd-item-img"
                data-testid="selectd-item-img"
              >
                <img
                  alt="image of Graphic Card"
                  src="src/assets/electronics/geforce-rtx-3080-ti.jpg"
                />
              </div>
              <div
                class="selected-item-info"
                data-testid="selected-item-name"
              >
                <p>
                  Graphic Card
                </p>
              </div>
              <div
                class="selected-item-price"
                data-testid="selected-item-price"
              >
                400
              </div>
            </div>
            <div
              class="upadate-no-of-item"
            >
              <div
                class="add"
              >
                <button
                  type="button"
                >
                  +
                </button>
              </div>
              <input
                class="no-of-item"
                name="no-of-item"
                type="text"
                value="1"
              />
              <div
                class="remove"
              >
                <button
                  type="button"
                >
                  -
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="cart"
    >
      <div
        class="close-cart"
        data-testid="close-cart"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
      <div
        id="cart-title"
      >
        <h1>
          <span
            class="block"
          >
            Your
          </span>
          <span
            class="block"
          >
            Shopping
          </span>
          <span
            class="block"
          >
            Bag
          </span>
        </h1>
      </div>
      <div
        id="show-items"
      >
        <div
          class="item"
          data-testid="show-items"
        >
          <div
            class="selected-item"
          >
            <div
              class="selectd-item-img"
              data-testid="selectd-item-img"
            >
              <img
                alt="image of Graphic Card"
                src="src/assets/electronics/geforce-rtx-3080-ti.jpg"
              />
            </div>
            <div
              class="selected-item-info"
              data-testid="selected-item-name"
            >
              <p>
                Graphic Card
              </p>
            </div>
            <div
              class="selected-item-price"
              data-testid="selected-item-price"
            >
              400
            </div>
          </div>
          <div
            class="upadate-no-of-item"
          >
            <div
              class="add"
            >
              <button
                type="button"
              >
                +
              </button>
            </div>
            <input
              class="no-of-item"
              name="no-of-item"
              type="text"
              value="1"
            />
            <div
              class="remove"
            >
              <button
                type="button"
              >
                -
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
